src/test/scala/io/iohk/ethereum/transactions/PendingTransactionsManagerSpec.scala

```diff
@@ -107,15 +108,46 @@ class PendingTransactionsManagerSpec extends FlatSpec with Matchers with ScalaFu
     pendingTxs.pendingTransactions.size shouldBe 0
   }
 
+  it should "override transactions with the same sender and nonce" in new TestSetup {
+    val firstTx = newStx(1, tx, keyPair1)
+    val otherTx = newStx(1, tx, keyPair2)
+    val overrideTx = newStx(1, tx.copy(value = 2 * tx.value), keyPair1)
+
+    pendingTransactionsManager ! AddOrOverrideTransaction(firstTx)
+    peerManager.expectMsg(PeerManagerActor.GetPeers)
+    peerManager.reply(Peers(Map(peer1 -> Handshaked)))
+
+    pendingTransactionsManager ! AddOrOverrideTransaction(otherTx)
+    peerManager.expectMsg(PeerManagerActor.GetPeers)
+    peerManager.reply(Peers(Map(peer1 -> Handshaked)))
+
+    pendingTransactionsManager ! AddOrOverrideTransaction(overrideTx)
+    peerManager.expectMsg(PeerManagerActor.GetPeers)
+    peerManager.reply(Peers(Map(peer1 -> Handshaked)))
+
+    val pendingTxs = (pendingTransactionsManager ? GetPendingTransactions).mapTo[PendingTransactionsResponse]
+      .futureValue.pendingTransactions
+
+    pendingTxs.map(_.stx) shouldEqual List(overrideTx, otherTx)
+
+    // overriden TX will still be broadcast to peers
```

Comment:
Because we do that immediately upon submission of a TX. We could for example introduce a timed delay, to give change for an override to discard the old TX before it gets broadcast, but do we want to go down this road?